http-transport-binding.md

```diff
@@ -0,0 +1,363 @@
+codes are non-normative illustrations.
+
+This specification also applies equivalently to HTTP/2 ([RFC7540][RFC7540]),
+which is compatible with HTTP 1.1 semantics.
+
+### 1.3. Content Modes
+
+This specification defines two content modes for transferring events:
+*structured* and *binary*. Every compliant implementation SHOULD support both
+modes.
+
+In the *structured* content mode, event metadata attributes and event data are
+placed into the HTTP request or response body using an [event
+format](#14-event-formats).
+
+In the *binary* content mode, the value of the event `data` attribute is placed
+into the HTTP request or response body as-is, with the `contentType` attribute
+value declaring its media type; all other event attributes are mapped to HTTP
+headers.
+
+### 1.4. Event Formats
+
+Event formats, used with the *stuctured* content mode, define how an event is
+expressed in a particular data format. All implementations of this
+specification MUST support the [JSON event format][JSON-format], but MAY
+support any additional, including proprietary, formats.
+
+### 1.5. Security
+
+This specification does not introduce any new security features for HTTP, or
+mandate specific existing features to be used. This specification applies
+identically to [HTTP over TLS]([RFC2818][RFC2818]).
+
+## 2. Use of CloudEvents Attributes
+
+This specification does not further define any of the [CloudEvents][CE] event
+attributes.
+
+Two of the event attributes, `contentType` and `data` are handled specially
+and mapped onto HTTP constructs, all other attributes are transferred as
+metadata without further interpretation.
+
+This mapping is intentionally robust against changes, including the addition
+and removal of event attributes, and also accommodates vendor extensions to the
+event metadata. Any mention of event attributes other than `contentType` and
+`data` is exemplary.
+
+### 2.1. contentType Attribute
+
+The `contentType` attribute is assumed to contain a [RFC2046][RFC2046]
+compliant media-type expression.
+
+### 2.2. data Attribute
+
+The `data` attribute is assumed to contain opaque application data that is
+encoded as declared by the `contentType` attribute.
+
+An application is free to hold the information in any in-memory representation
+of its choosing, but as the value is transposed into HTTP as defined in this
+specification, the assumption is that the `data` attribute value is made
+available as a sequence of bytes.
+
+For instance, if the declared `contentType` is
+`application/json;charset=utf-8`, the expectation is that the `data` attribute
+value is made available as [UTF-8][RFC3629] encoded JSON text to HTTP.
+
+## 3. HTTP Message Mapping
+
+The event binding is identical for both HTTP request and response messages.
+
+The content mode is chosen by the sender of the event, which is either the
+requesting or the responding party. Gestures that might allow solicitation of
```

Comment:
What's a "gesture?"

Reply:
I paused on that one too  :-) I think he meant "there may be data elsewhere" - but that's just a guess.

Reply:
We can reword that. With "gesture" I generally refer to an protocol interaction pattern. "Using a GET request to obtain an event held in a buffer". That's broader than "method".